main.cpp

```diff
@@ -204,21 +191,7 @@ IntType& IntType::operator-=(const int other)
 
 IntType& IntType::operator/=( const int other) 
 {
-    if (other == 0)
-    {
-        std::cout << "error divide by zero" << std::endl;
-        *intTypeHeap = 0; 
-    } 
-    else 
-    { 
-        *intTypeHeap /= other;
-    }
-    return *this;
```

Comment:
I did not say to delete this.   I said don't set `intTypeHeap` to 0 if `other == 0`.